Kudu.Core/Permissions/PermissionHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+using Kudu.Contracts.Permissions;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Deployment;
+using Kudu.Core.Deployment.Generator;
+using Kudu.Core.Infrastructure;
+
+namespace Kudu.Core.Permissions
+{
+    /// <summary>
+    /// This handler is meant for Linux enviroment only
```

Comment:
Another idea is to do it lazily when needed.   For instance, right before we [fetch] (https://github.com/projectkudu/kudu/blob/6886065d11e1f3b37f52dcc65a69c7eb4de3d95d/Kudu.Services/ServiceHookHandlers/FetchHandler.cs#L238) or with the fetch itself.   You will have everything needed in the context.

I think this is the only place we actually use SSH.